src/azul/service/repository_controller.py

```diff
@@ -66,6 +72,17 @@ def download_file(self,
             assert file_version is not None
             assert file_name is not None
 
+        # To avoid a `416 Range Not Satisfiable` response to curl clients
+        # using the `--continue-at -` option, return an empty 206 response
+        # if the requested byte range starts at the end of the file.
+        if is_completed_range_request:
+            return {
+                'Status': 206,
+            }
+
+        plugin = self.repository_plugin(catalog)
+        download_cls = plugin.file_download_class()
```

Comment:
Opinion: Not too confident that this move is necessary. Only because it increases the diff of the PR a bit. But I also see the value in grouping assignations based on their relevance with neighboring assignations and usages.

Reply:
Agreed. The re-arrangement is unrelated to this ticket. I'll revert the move.